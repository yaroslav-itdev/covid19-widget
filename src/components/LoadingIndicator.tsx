import React from 'react'
import Loader from 'react-loader-spinner'
import {makeStyles} from "@material-ui/styles";

export default () => {
    const classes = useStyles();

    return (
        <div className={classes.wrapper}>
            <Loader
                type="Grid"
                color="#00BFFF"
                height={100}
                width={100}
            />
        </div>
    )
}

const useStyles = makeStyles({
    wrapper: {
        display: 'flex',
        alignItems: 'center',
        height: 400,
    },
});
