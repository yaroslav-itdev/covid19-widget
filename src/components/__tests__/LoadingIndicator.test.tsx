import React from 'react'
import renderer from 'react-test-renderer'
import LoadingIndicator from '../LoadingIndicator'

describe('LoadingIndicator', () => {
    it('should renders without errors and matches snapshot', () => {
        const tree = renderer.create(<LoadingIndicator />)

        expect(tree.toJSON()).toBeTruthy()
        expect(tree).toMatchSnapshot()
    })
})
