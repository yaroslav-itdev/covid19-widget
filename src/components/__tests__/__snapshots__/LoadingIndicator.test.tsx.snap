// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator should renders without errors and matches snapshot 1`] = `
<div
  className="makeStyles-wrapper-1"
>
  <div
    aria-busy="true"
    className=""
    style={Object {}}
  >
    <svg
      aria-label="audio-loading"
      fill="#00BFFF"
      height={100}
      viewBox="0 0 105 105"
      width={100}
    >
      <circle
        cx="12.5"
        cy="12.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="52.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="100ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="12.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="300ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="52.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="600ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="12.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="800ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="52.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="400ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="12.5"
        cy="92.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="700ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="52.5"
        cy="92.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="500ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
      <circle
        cx="92.5"
        cy="92.5"
        r={12.5}
      >
        <animate
          attributeName="fill-opacity"
          begin="200ms"
          calcMode="linear"
          dur="1s"
          repeatCount="indefinite"
          values="1;.2;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
